File: SOLID principles/4-Interface-segregation-principle/src/main/java/solution/service/OrderPersistenceService.java
package solution.service;

import solution.entity.Order;

import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

//Stores Order entities
public class OrderPersistenceService implements PersistenceService<Order> {
	
	private static final Map<Long, Order> ORDERS = new HashMap<>();
	
	@Override
	public void save(Order entity) {
		synchronized (ORDERS) {
			ORDERS.put(entity.getId(), entity);
		}
	}

	@Override
	public void delete(Order entity) {
		synchronized (ORDERS) {
			ORDERS.remove(entity.getId());
		}
	}

	@Override
	public Order findById(Long id) {
		synchronized (ORDERS) {
			return ORDERS.get(id);
		}
	}

}

File: SOLID principles/4-Interface-segregations-principle/dotNET/dotNET/Solution/service/OrderPersistenceService.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace dotNET.Solution.service
{
    class OrderPersistenceService : PersistenceService<Order>
    {
        private Dictionary<long, Order> ORDER = new Dictionary<long, Order>();

        public void delete(Order entity)
        {
            ORDER.Remove(entity.Id);
        }

        public Order findById(long id)
        {
            Order Order;
            ORDER.TryGetValue(id, out Order);
            return Order;
        }

        public void save(Order entity)
        {
            ORDER.Add(entity.Id, entity);
        }
    }
}
